presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/TableScanMatcher.java

```diff
@@ -39,9 +50,11 @@ public boolean matches(PlanNode node, Session session, Metadata metadata, Symbol
             TableScanNode tableScanNode = (TableScanNode) node;
             TableMetadata tableMetadata = metadata.getTableMetadata(session, tableScanNode.getTable());
             String actualTableName = tableMetadata.getTable().getTableName();
-            if (expectedTableName.equalsIgnoreCase(actualTableName)) {
-                return true;
+            boolean matches = expectedTableName.equalsIgnoreCase(actualTableName);
+            if (matches && predicate.isPresent() && tableScanNode.getOriginalConstraint() != null) {
```

Comment:
Thinking about this more, the problem is that pushdown into TableScan is a function of each specific connector, so testing the layout predicate or current constraint will produce different results for different connectors.